site/propsGen.js

```diff
@@ -0,0 +1,32 @@
+const docgen = require("react-docgen-typescript");
+const path = require("path");
+const { writeFile } = require("fs");
+
+const componentPackages = ["core", "data-grid", "icons"]; // Adding more packages here will generate additional prop files
+
+const componentFiles = componentPackages.map((package) =>
+  path.join(__dirname, "..", "packages", package, "src", "index.ts")
+);
+
+const options = {
+  propFilter: (prop) =>
+    !/@types[\\/]react[\\/]/.test(prop.parent?.fileName || ""),
+};
+
+const components = componentFiles.map((component) =>
+  docgen.parse(component, options)
+);
+
+const filePaths = componentPackages.map(
+  (package) => `./src/props/${package}-props.json`
+);
+
+components.forEach((component, index) =>
+  writeFile(filePaths[index], JSON.stringify(component, null, 2), (error) => {
```

Comment:
This currently fails because the `props/` dir does not exist in freshly checked out git repo. (it probably works fine on your machine as you probably added that dir manually at some point while working on this)

I suggest you...

* Either check a dummy file into that `props/` dir into git, so that it's guaranteed to exist. E.g. an empty file like `props/.gitkeep`, or maybe put a `README.md` in there that explains what the purpose of that dir is.
* Or add some logic here into `propGen.js` to check if the dir exists and, if not, create it first.

Reply:
Good spot, I've added a `props` dir with a `README` file